### ISSUE TITLE
.gitignore specific to subrepo

### ISSUE THREAD

**Issue by fherreazcue** [ingydotnet/git-subrepo#596]
Is it possible to ignore files for the subrepo commits/pushes, but not ignore them in the main repo?

I am using git subrepo to re-use code that generates files based on some templates. The templates live in the subrepo (S), and the subrepo lives in multiple bigger repos (R1, R2...) with different directory structures. 
The workflow is generally to pull S from R*, run a script (which also lives in S) to generate a bunch of R* specific files, and then push these files to R*.
This would be fine if S was "read only", but if you ever try to push to S, all of the R* specific files are pushed, and eventually cloned into the other R* repos. I want to be able to ignore those files in S commits, say by having a .gitsubrepoignore file.

I found a  couple of issues (#437 & #64) that touch on a feature I could not find how to work, and I believe those issues refer more to ignoring files when pulling a subrepo.

**Comment by admorgan** [comment]
No, git subrepo does not have a mechanism to ignore files other than .gitignore. Therefore it would not be possible to have files committed to the S directory in R but not pushed to S. This is also a feature I am not interested in implementing. I always look at patches, but I am not inclined to be excited about this one.

I obviously don't have any information about your projects, but I would store the output in a directory in R that does not overlap the S directory. Then you have the tools to generate them, but not trying to filter the output. To be honest in my work projects I don't allow committing the output of something that can be compiled (I count scripts that generate output of any sort as compiling).

**Comment by ryan-ph** [comment]
Sorry to comment on a closed issue, but I have a specific use-case in mind.

I have a monorepo, A, that is built using a language-agnostic build manager (e.g. bazel, buck2) but also supports building subrepos using the build tool packaged with the language (e.g. cargo). The root of A contains build manager configuration and then each subdirectory contains `BUILD` files committed to the code base that specify the build targets. I would like to push the subrepos to their own reponsitories which have no visibility of the build manager while excluding the `BUILD` files on the push as they are useless outside of A.

In this case, the `BUILD` files aren't artifacts or outputs and it's not easy to move them completely out of the subrepos. Happy to try to work on a patch if this feature is something you'd be willing to review and add.

**Comment by admorgan** [comment]
I don't mind comments on closed issues, if they are persuasive I can always change my mind. and open it back up.

I have a similar system that uses subrepos. In my case I keep those build files outside of the subrepo directory. Say I was using zlib in my project as a subrepo I would have

project/zlib/BUILD
project/zlib/lib

Where build had the instructions to build zlib and the actual subrepo was project/zlib/lib.

Other projects I have worked with keep all the build files together

project/BUILD.zlib
project/zlib

In either case this is easier for subrepo to work with because changes to the BUILD file which isn't managed by subrepo but is in a directory managed by subrepo gets really complicated, and our branch rewrite rules are already really complicated and hard to verify.

I am not saying I wouldn't accept a PR with this feature, I am saying the amount of testing that I would require before accepting that PR would be extremely extensive. If you would like to work on this let me know and I will reopen this issue, but I am not personally adding it to my TODO list.
